masterdata/returnRequest/schema.json

```
@@ -0,0 +1,212 @@
+{
+  "properties": {
+    "orderId": {
+      "type": "string",
+      "required": true
+    },
+    "totalReturnAmount": {
+      "type": "integer",
+      "required": true
+    },
+    "sequenceNumber": {
+      "type": "string",
+      "required": true
+    },
+    "status": { "$ref": "#/$defs/status", "required": true },
+    "returnTotals": {
+      "type": "array",
+      "required": true,
+      "items": {
+        "type": "object",
+        "properties": {
+          "id": {
+            "type": "string",
+            "required": true,
+            "enum": ["items", "shipping", "tax"]
+          },
+          "value": {
+            "type": "integer",
+            "required": true
+          }
+        }
+      }
+    },
+    "customerProfileData": {
+      "type": "object",
+      "required": true,
+      "properties": {
+        "userId": {
+          "type": "string",
+          "required": true
+        },
+        "name": {
+          "type": "string"
+        },
+        "email": {
+          "type": "string",
+          "format": "email"
+        },
+        "phoneNumber": { "type": "string", "maxLength": 50 }
+      }
+    },
+    "pickupReturnData": {
+      "type": "object",
+      "required": true,
+      "properties": {
+        "addressId": {
+          "type": "string"
+        },
+        "address": {
+          "type": "string",
+          "required": true
+        },
+        "city": {
+          "type": "string",
+          "required": true
+        },
+        "state": {
+          "type": "string",
+          "required": true
+        },
+        "country": {
+          "type": "string",
+          "required": true
+        },
+        "zipCode": {
+          "type": "string",
+          "required": true
+        },
+        "addressType": {
+          "type": "string",
+          "required": true,
+          "enum": ["pickup-point", "customer-address"]
+        },
+        "returnLabel": {
+          "type": "string"
+        }
+      }
+    },
+    "refundPaymentData": {
+      "type": "object",
+      "required": true,
+      "properties": {
+        "refundPaymentMethod": {
+          "type": "string",
+          "required": true,
+          "enum": ["bank", "card", "giftCard"]
+        },
+        "iban": {
+          "type": "string"
+        },
+        "accountHolderName": {
+          "type": "string"
+        },
+        "paidWithCc": {
+          "type": "boolean"
+        }
+      }
+    },
+    "items": {
+      "type": "array",
+      "required": true,
+      "items": {
+        "type": "object",
+        "properties": {
+          "orderItemIndex": { "type": "integer", "required": true },
+          "quantity": { "type": "integer", "required": true },
+          "verifiedItems": { "type": ["integer", "null"], "required": true },
+          "reasonCode": { "type": "string", "required": true },
+          "reason": { "type": "string", "required": true },
```

Changing the way to handle reason to be more semantic. `otherReason` will be a free text user can enter in the UI. As now, it's saved as `reason`, which can be confusing.

```suggestion
          "returnReason": {
            "type": "object",
            "required": true,
            "properties": {
              "reason": {
                "type": "string",
                "required": true
              },
              "otherReason": {
                "type": "string"
              }
            }
          },
```